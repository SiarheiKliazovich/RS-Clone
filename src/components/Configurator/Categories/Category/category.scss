@import "../../../../sass/variables";
@import "../../../../sass/mixins";

.category__list_item {
  width: calc(33.3% - 15px);
  padding: 15px;
  background: $color-grey-tie;
  border-radius: 5px;
  margin: 7px;
  overflow: hidden;

  @include media-mobile {
    padding: 0;
    margin: 10px 10px;
    width: 65%;
  }
}

.category-blok {
  color: $color-white1;
  display: block;
  text-decoration: none;
  padding: 15px 15px;
}


.category-img {
  width: 100%;
  height: auto;
}

.category-name {
  color: $color-white1;
  padding: 4px 10px;
  line-height: 1.5;
  border-radius: 5px;
  cursor: pointer;
  border: none;

  &:hover {
   opacity: 0.8;
  }
}