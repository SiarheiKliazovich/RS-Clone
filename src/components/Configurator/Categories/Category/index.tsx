import { FunctionComponent } from "react";
import { CategoryType } from "../../../../types/configurator";
import { nightTheme } from "../../../../data/constants";
import "./category.scss";

const Category: FunctionComponent<CategoryType> = ({
  tie,
  setType,
  accentColor,
  isNavbarNightMode,
}: CategoryType) => {
  const backgroundColor = nightTheme.background.element;

  return (
    <li className="category__list_item">
      <div className="category-blok">
        <img className="category-img" src={tie.image} alt="img-ties" />

        <button
          style={{
            backgroundColor: isNavbarNightMode
              ? backgroundColor
              : accentColor.static,
          }}
          className="category-name"
          onClick={() => setType(tie.type)}
        >
          {tie.categoryName}
        </button>
      </div>
    </li>
  );
};

export default Category;
