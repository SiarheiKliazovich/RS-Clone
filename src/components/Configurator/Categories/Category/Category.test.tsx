import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Category from ".";

describe("Active", () => {
  test("renders Category component correctly", () => {
    const props = {
      tie: {
        id: 1,
        image: "tie-image-url",
        type: "some-tie-type",
        categoryName: "Some Category",
        price: "30",
      },
      accentColor: { static: "#000", hover: "#fff" },
      isNavbarNightMode: false,
    };

    const { container } = render(
      <Router>
        <Category {...props} />
      </Router>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
