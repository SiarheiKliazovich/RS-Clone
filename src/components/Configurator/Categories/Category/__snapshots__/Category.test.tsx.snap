// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active renders Category component correctly 1`] = `
<li
  class="category__list_item"
>
  <div
    class="category-blok"
  >
    <img
      alt="img-ties"
      class="category-img"
      src="tie-image-url"
    />
    <a
      class="category-name"
      href="/configurator/some-tie-type"
      style="background-color: rgb(0, 0, 0);"
    >
      Some Category
    </a>
  </div>
</li>
`;
